share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Are we targeting portable code that uses the pthread API — i.e., does not depend on platform specific pthread API headers? (I suppose we are.)

Is `pthreadtypes.h` part of the pthread API specification? (I suppose it is not.)

If our `pthread.h` header were being picked up by Clang, instead of the default system `pthread.h` header, then how would it be possible for “other header[s] [to] recursively include the [system] pthread typedefs?”

Something does not make sense here.